Linux/tree-common/init

```diff
@@ -249,6 +249,14 @@ log_begin_msg "Creating default configuration files on rootfs..."
 run mkdir -p "${rootmnt}/etc"
 run touch ${rootmnt}/etc/hostname ${rootmnt}/etc/resolv.conf
 [ -s ${rootmnt}/etc/hostname ] || run sh -ec "echo ${HOSTNAME} > ${rootmnt}/etc/hostname"
+if [ ! -f ${rootmnt}/etc/hosts.bak ]; then
+    cp ${rootmnt}/etc/hosts ${rootmnt}/etc/hosts.bak
+
+    FQDN=$(echo $HOSTNAME | cut -d'.' -f1)
```

Comment:
Ok, the variable names are confusing. Your FQDN is not storing the fully qualified domain name, it's storing just the first part.